### PR TITLE
Add FXIOS-4467 [v104] Epic branch bookmark folders changes + refactor of panels toolbar

### PR DESCRIPTION
**Pull request by lmarceau**
# [FXIOS-4467](https://mozilla-hub.atlassian.net/browse/FXIOS-4467) https://github.com/mozilla-mobile/firefox-ios/issues/11140
This PR is for the bookmark epic branch which bundles following PRs (in order they were made pointing to this epic-branch):
- https://github.com/mozilla-mobile/firefox-ios/pull/11168
- https://github.com/mozilla-mobile/firefox-ios/pull/11225
- https://github.com/mozilla-mobile/firefox-ios/pull/11287
- https://github.com/mozilla-mobile/firefox-ios/pull/11334
- https://github.com/mozilla-mobile/firefox-ios/pull/11354
- https://github.com/mozilla-mobile/firefox-ios/pull/11362
- https://github.com/mozilla-mobile/firefox-ios/pull/11376

I've asked our QA to retest this branch (after fixes were made for #11323, #11324, #11325, #11326, #11327, #11328) so we can get a green ✅ to merge to `main`. This can be tested with [Build #14029](https://app.bitrise.io/build/fa3ac935-09c8-46b0-88a3-fbd1171c160a).

### CODE REVIEW

**Review by electricRGB — APPROVED** [review]
Given that this is an epic branch, and all the other PRs have been approved, I approve. :P

**Review by thatswinnie — APPROVED** [review]

**Comment by lmarceau** [comment]
I'll wait for QA approval before merging

**Comment by lmarceau** [comment]
Merging as this will be part of testing for v104

**Pull request merged**
